File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.java
class Solution {
    public int[] findThePrefixCommonArray(int[] A, int[] B) {
        int n=A.length;
        int[] res=new int[n];
        int[] cnt=new int[51];
        for (int i=0;i<n;i++){
            cnt[A[i]]++;
            cnt[B[i]]++;
            int newCommon=0;
            if (cnt[A[i]]==2){
                newCommon++;
            }
            if (A[i]!=B[i] && cnt[B[i]]==2){
                newCommon++;
            }
            res[i]=newCommon;
            if(i>0){
                res[i]+=res[i-1];
            }
        }
        return res;

    }
}

File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        maxCommon=0
        res=[0]*len(A)
        for i in range(len(A)):
            new_a=A[:i+1]
            new_b=B[:i+1]
            temp=0
            for x in new_a:
                if x in new_b:
                    temp+=1
            res[i]=temp
                    
        return res